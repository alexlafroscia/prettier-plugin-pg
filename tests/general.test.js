const prettier = require("prettier");

function check(code) {
  test(code, () => {
    const formattedCode = prettier.format(code, {
      parser: "postgresql-sql",
      plugins: ["./"],
    });
    expect(formattedCode).toMatchSnapshot();
  });
}

check(`select 1;
select 2; create function foo() returns text as $$ select 'hi'; $$ language sql;`);
check(`select 1;`);
check("sEleCt 1;");
check("sEleCt 1 from my_table;");
check(`sEleCt 1 from "table";`);
check(`sEleCt 1 from "Table";`);
check(`sEleCt     1     as       one     from     "Table";`);
check(`sEleCt  
1     as       
       one     from     "Table"
       ;`);
