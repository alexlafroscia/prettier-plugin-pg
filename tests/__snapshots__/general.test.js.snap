// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sEleCt  
1     as       
       one     from     "Table"
       ; 1`] = `
"SELECT 1 AS one FROM \\"Table\\";
"
`;

exports[`sEleCt     1     as       one     from     "Table"; 1`] = `
"SELECT 1 AS one FROM \\"Table\\";
"
`;

exports[`sEleCt 1 from "Table"; 1`] = `
"SELECT 1 FROM \\"Table\\";
"
`;

exports[`sEleCt 1 from "table"; 1`] = `
"SELECT 1 FROM \\"table\\";
"
`;

exports[`sEleCt 1 from my_table; 1`] = `
"SELECT 1 FROM my_table;
"
`;

exports[`sEleCt 1 from "my_table"; 1`] = `
"SELECT 1 FROM my_table;
"
`;

exports[`sEleCt 1; 1`] = `
"SELECT 1;
"
`;

exports[`select 1;
select 2; create function foo() returns text as $$ select 'hi'; $$ language sql; 1`] = `
"SELECT 1;
SELECT 2;
CREATE FUNCTION foo() RETURNS TEXT LANGUAGE SQL AS $$ SELECT 'hi'; $$;
"
`;

exports[`select 1; 1`] = `
"SELECT 1;
"
`;
